src/Coordinate.java

```diff
@@ -8,39 +8,56 @@ public Coordinate() {
 		this.hasShip = false;
 	}
 
-	// TODO
-	// call the coordinate, check if the ship was hit, missed, or sunk
-	// read about the observer pattern
-	//
 	public String call() throws Exception {
+		
 		if (!this.called) {
+			
 			this.notifyObserver();
+			this.called = true;
+			
+			if (this.hasShip) {
+				this.observer.hit();
+			}
+			
 			return "Miss";
+			
 		} else {
+			
 			throw new Exception("Coordinate already called");
+			
 		}
 	}
 
-	// TODO
-	//	read about inheritance and calling parent
-	//	methods in Java
-	//
 	@Override
 	public void attach(Ship observer) {
-		// TODO
+		
+		this.hasShip = true;
+		this.observer = observer;
```

Comment:
this isn't the observer pattern or good OO. See calling the parent class methods: `super.attach()`